packages/core/src/use-query-hooks.ts

```diff
@@ -49,14 +50,15 @@ export function useQuery<T extends QueryFn, TResult = PromiseReturnType<T>>(
     throw new Error("useQuery is missing the first argument - it must be a query function")
   }
 
-  const suspense =
-    options?.enabled === false || options?.enabled === null ? false : options?.suspense
+  const legacyMode = getConfig?.().experimental?.reactMode === "legacy"
```

Comment:
@MrLeebo as of nextjs 10.2 (already on our canary), it's now `experimental.reactRoot = true/false`